test/development/basic/next-rs-api.test.ts

```diff
@@ -87,14 +88,41 @@ function raceIterators<T>(iterators: AsyncIterableIterator<T>[]) {
   })()
 }
 
+async function drainAndGetNext<T>(
+  stream: AsyncIterableIterator<TurbopackResult<T>>
+) {
+  const next = stream.next()
+  while (true) {
+    const result = await Promise.race([
+      new Promise((r) => setTimeout(() => r({ next }), 100)),
+      next.then(() => undefined),
+    ])
+    if (result) return result
+  }
+}
```

Comment:
Comment please. I don't understand what this is trying to do. We're pulling a value from the async iterator, but then discarding it if it resolves? And returning the unresolved promise if it doesn't resolve?